src/strategies/DSL/actions/limitSwapAction.ts

```diff
@@ -1,17 +1,59 @@
-import { LimitSwapActionArgs, PrimitiveArgs } from '@brinkninja/types'
+import { LimitSwapActionArgs, PrimitiveArgs, PrimitiveParamValue } from '@brinkninja/types'
+import { convertToX96HexPrice, toBigInt, toTokenArgs } from '@brink-sdk/internal';
+
+const FLAT_PRICE_CURVE_ADDRESS = '0xC509733B8dDdbab9369A96F6F216d6E59DB3900f';
+
+export class InvalidInputError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = 'InvalidInputError';
+    }
+}
 
 function limitSwapAction ({
+  id,
   owner,
   tokenIn,
   tokenOut,
   tokenInAmount,
-  tokenOutAmount,
-  price
+  tokenOutAmount
 }: LimitSwapActionArgs): PrimitiveArgs[] {
+  if (tokenOutAmount === undefined) {
+    throw new InvalidInputError('tokenOutAmount is required')
+  }
+
+  let bigTokenInAmount: bigint
```

Comment:
just a suggestion, but I usually name like `tokenInAmountBN` if trying to distinguish between a non bigint and bigint typed var

Reply:
done